lightning/src/ln/channelmanager.rs

```diff
@@ -4001,7 +4001,10 @@ where
 	/// The invoice's `payer_metadata` is used to authenticate that the invoice was indeed requested
 	/// before attempting a payment. [`Bolt12PaymentError::UnexpectedInvoice`] is returned if this
 	/// fails or if the encoded `payment_id` is not recognized. The latter may happen once the
-	/// invoice is paid and the payment no longer tracked.
+	/// payment is no longer tracked because the payment was attempted after:
+	/// - an invoice for the `payment_id` was already paid,
+	/// - one timer tick has elapsed since initially requesting the invoice when paying an offer, or
```

Comment:
Hmm... timer ticks are very much part of the public API. I could link to either the "Operation" section of the `ChannelManager` docs or directly to `timer_tick_occurred` docs. Will add some clarification about what to do in case of a failure.